### PR TITLE
[quicklook] Update for Xcode11 beta 1-3

### PR DESCRIPTION
**Pull request by whitneyschmidt**
Changes for b1-3, no updates from b3 according to wiki page.

### CODE REVIEW

File: src/quicklook.cs [hunk 1]
```diff
@@ -119,6 +128,19 @@ interface QLPreviewControllerDelegate {
 		[Export ("previewController:transitionViewForPreviewItem:"), DelegateName ("QLTransitionView"), DefaultValue (null)]
 		[return: NullAllowed]
 		UIView TransitionViewForPreviewItem (QLPreviewController controller, IQLPreviewItem item);
+
+		[iOS (13,0)]
+		[Export ("previewController:editingModeForPreviewItem:"), DelegateName("QLEditingMode"), DefaultValue ("QLPreviewItemEditingMode.Disabled")]
+		QLPreviewItemEditingMode EditingModeForPreviewItem (QLPreviewController controller, IQLPreviewItem item);
```
Comment (spouliot) [inline src/quicklook.cs]: prefix: use a verb for methods, e.g. `Get...`
suffix: avoid repeating parameter names, e.g. drop `ForPreviewItem`

so `GetEditingMode`

note: ObjC selectors are `{method_name}{param1}:{param2}` - i.e. there's no `:` between the method name and the **first** parameter name. In doubt check the name used by Apple for the Swift API.

File: src/quicklook.cs [hunk 2]
```diff
@@ -119,6 +128,19 @@ interface QLPreviewControllerDelegate {
...
+
+		[iOS (13,0)]
+		[Export ("previewController:didUpdateContentsOfPreviewItem:"), EventArgs ("QLPreviewControllerDelegateDidUpdate")]
+		void DidUpdateContentsOfPreviewItem (QLPreviewController controller, IQLPreviewItem item);
```
Comment (spouliot) [inline src/quicklook.cs]: same -> `DidUpdateContents`

File: src/quicklook.cs [hunk 3]
```diff
@@ -119,6 +128,19 @@ interface QLPreviewControllerDelegate {
...
+
+		[iOS (13,0)]
+		[Export ("previewController:didSaveEditedCopyOfPreviewItem:atURL:"), EventArgs ("QLPreviewControllerDelegateDidSave")]
+		void DidSaveEditedCopyOfPreviewItem (QLPreviewController controller, IQLPreviewItem item, NSUrl modifiedContentsUrl);
```
Comment (spouliot) [inline src/quicklook.cs]: same -> `DidSaveEditedCopy`

File: src/quicklook.cs (hunk 1 above)
Comment (rolfbjarne) [reply src/quicklook.cs]: This also means the parameter can be better named as `previewItem`:

```csharp
QLPreviewItemEditingMode GetEditingMode (..., IQLPreviewItem previewItem);
```

File: src/quicklook.cs (hunk 2 above)
Comment (rolfbjarne) [reply src/quicklook.cs]: same -> `IQLPreviewItem previewItem`

File: src/quicklook.cs (hunk 3 above)
Comment (rolfbjarne) [reply src/quicklook.cs]: same -> `IQLPreviewItem previewItem`

File: src/quicklook.cs (hunk 1 above)
Comment (whitneyschmidt) [reply src/quicklook.cs]: Updated this and others! Thanks @spouliot and @rolfbjarne :)

File: src/quicklook.cs [hunk 4]
```diff
@@ -119,6 +128,19 @@ interface QLPreviewControllerDelegate {
 		[Export ("previewController:transitionViewForPreviewItem:"), DelegateName ("QLTransitionView"), DefaultValue (null)]
 		[return: NullAllowed]
 		UIView TransitionViewForPreviewItem (QLPreviewController controller, IQLPreviewItem item);
+
+		[iOS (13,0)]
+		[Export ("previewController:editingModeForPreviewItem:"), DelegateName("QLEditingMode"), DefaultValue ("QLPreviewItemEditingMode.Disabled")]
```
Comment (tj-devel709) [inline src/quicklook.cs]: How did you know to use the DelegateName and DefaultValue attributes and their values?

Comment (whitneyschmidt) [reply src/quicklook.cs]: Looking at documentation and other similar code - ie, transitionViewForPreviewItem has DelegateName + DefaultValue. Then asking Alex :)

**Pull request merged**